src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -199,7 +199,7 @@ private string GetVersion(string assetId)
         {
             var buildsManifestMetadata = new List<BuildData>();
 
-            foreach (string manifestPath in Directory.GetFiles(manifestsFolderPath))
+            foreach (string manifestPath in Directory.GetFiles(manifestsFolderPath, "*.xml", SearchOption.AllDirectories))
```

Comment:
Is it correct to look at AllDirectories, and was it a bug that we previously didnt? The issue that this is fixing is lacking details to know what was broken, and how this would fix that situation.

Reply:
Thanks for checking this. AFAIU there is no problem in checking nested directories, the path searched is specifically for storing manifest files. The issue being addressed here is to prevent the task from parsing non-xml files that the user might mistakenly put on the same directory as the manifests.